keras/callbacks.py

```diff
@@ -776,7 +781,7 @@ def on_epoch_end(self, epoch, logs=None):
                     feed_dict = dict(zip(tensors, batch_val))
                     result = self.sess.run([self.merged], feed_dict=feed_dict)
                     summary_str = result[0]
-                    self.writer.add_summary(summary_str, epoch)
+                    self.writer.add_summary(summary_str, self.seen)
```

Comment:
If write batch performance is false self.seen should probably be equal to the epoch so the current behavior remains unchanged.